src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -1659,14 +1657,15 @@ bool SharedRuntime::handle_ic_miss_helper_internal(Handle receiver, CompiledMeth
   return true;
 }
 
-methodHandle SharedRuntime::handle_ic_miss_helper(JavaThread *thread, TRAPS) {
+methodHandle SharedRuntime::handle_ic_miss_helper(TRAPS) {
+  JavaThread* thread = THREAD->as_Java_thread();
```

Comment:
Everything goes back to the JRT macros where THREAD is created from thread, so they are always the same current thread instance.

Reply:
Have you check all possible call paths? Could one of these functions be called from a place other than the JRT functions?

Reply:
This is why THREAD is clearly the current thread because that's what the JRT_/JVM_ macros do.  Just use THREAD as the current thread for ResourceMark and whatever else needs it below.